app/validators/element_aviti_validator.rb

```diff
@@ -5,10 +5,31 @@ def self.validation_info
     '300PE (MO, HO) require only one request.'
   end
 
-  # Adds an error to the record if more than one request exists with a read length of 300.
+  # Validates that a batch contains the correct number of requests based on read length.
+  #
+  # - If any request has a read length of 300, the batch must contain exactly one request.
+  # - Otherwise, the batch must contain exactly two requests.
+  #
+  # Adds errors to the record if these conditions are not met.
   def validate(record)
-    return if record.requests.one? || record.requests.none? { |r| r.request_metadata.read_length == 300 }
+    if record.requests.any? { |r| r.request_metadata.read_length == 300 }
+      validate_single_request_for_read_length300(record)
+    else
+      validate_exactly_two_requests(record)
+    end
+  end
+
+  private
+
+  def validate_single_request_for_read_length300(record)
+    return unless record.requests.size > 1
 
     record.errors.add(:base, 'Batches can contain only one request when the read length is 300')
   end
+
+  def validate_exactly_two_requests(record)
+    return unless record.requests.size != 2
```

Comment:
Can this be written as `return if record.requests.size == 2` instead, to avoid double negative?